src/Type/Php/PregSplitDynamicReturnTypeExtension.php

```diff
@@ -36,17 +47,131 @@ public function isFunctionSupported(FunctionReflection $functionReflection): boo
 
 	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): ?Type
 	{
-		$flagsArg = $functionCall->getArgs()[3] ?? null;
+		$args = $functionCall->getArgs();
+		if (count($args) < 2) {
+			return null;
+		}
+		$patternArg = $args[0];
+		$subjectArg = $args[1];
+		$limitArg = $args[2] ?? null;
+		$flagArg = $args[3] ?? null;
+		$patternType = $scope->getType($patternArg->value);
+		$patternConstantTypes = $patternType->getConstantStrings();
+		$subjectType = $scope->getType($subjectArg->value);
+		$subjectConstantTypes = $subjectType->getConstantStrings();
```

Comment:
I would move the subject type declartion below the early returning if in the next block